dqnroute/simpy_router.py

```diff
@@ -0,0 +1,57 @@
+import simpy
+import random
+
+class SimPyAbstractRouter(object):
+    def __init__(self, env, id):
+        self.env = env
+        self.id = id
+        
+        self.send_process = env.process(self.sendPkgProcess())
+        self.get_process = env.process(self.getPkgProcess())
+        self.send_event = env.event()
+        self.get_event = env.event()
+
+    def setNeighbours(self, neighbours):
+        self.nghbrs = neighbours
+        
+    def sendPkgProcess(self):
+        while True:
+            pkg, dest = yield self.send_event
+            destRouter = self.nghbrs[dest]
+            print("Sending packet: {} -> {} at time {}...".format(self.id, destRouter.id, self.env.now))
+            #actual sending simulation
+            
+            #print("Start sending at: ", self.env.now)
+            yield self.env.timeout(random.randint(10, 20))
```

Comment:
AFAIU, this means that the next package would not be sent before the previous package arrives. Not good. The router should be able to receive and send a lot of packages in a row. I think that instead of yielding timeout event you should manually set a callback event for it.